### ISSUE TITLE
Stop showing discardable keys like created_by

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/iD#9854]
### URL

https://www.openstreetmap.org/edit?editor=id&node=289136538#map=19/50.04234/20.04962

### How to reproduce the issue?

See https://www.openstreetmap.org/edit?editor=id&node=289136538#map=19/50.04234/20.04962

Or http://overpass-turbo.eu/s/1z5O for more test cases

### Screenshot(s) or anything else?

triggered by https://community.openstreetmap.org/t/polska-propozycja-automatycznego-usuniecia-klucza-created-by/102601/4 (which proposed bot edit to mass remove `created_by` after noticing it in iD)

![screen03](https://github.com/openstreetmap/iD/assets/899988/32c342bb-cf54-4656-baa8-e73a98fade80)

Note that JOSM is hiding it from user and not showing at all.

From https://github.com/openstreetmap/iD/issues/7338 it seems that iD is displaying it and removing on saving.

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Firefox 116.0.2 (64-bit)

**Comment by AnitaMari** [comment]
Hi! I want to try to fix the issue. It's the first time I'm contributing to an open-source project, so I hope to be able to fix it and to really contribute in something. And to know how to do it.

Just to clarify, what I need to do is to remove "created_by JOSM" that appears in the edit feature?
 
![screenshot1](https://github.com/openstreetmap/iD/assets/116835304/d8d38c2a-7d82-424b-8dc7-d671ad517b2a)

**Comment by tordans** [comment]
@AnitaMari I don't think there is agreement on how to best solve this issue, yet.

On way to solve this would be to add `created_by` to this file https://github.com/openstreetmap/id-tagging-schema/blob/main/data/deprecated.json which would trigger an update dialogue to remove the key.

IMO the "Tags" section should always show all tags. So in case you where planning to hide it in this UI, I don't think that would be a good solution.

Update: The deprecation was done already https://github.com/openstreetmap/iD/issues/7338. And it is in line with what the wiki says https://wiki.openstreetmap.org/wiki/Key:created_by So I wonder what this issue is actually about…

**Comment by tordans** [comment]
@tyrasd I now see the "[good first issue](https://github.com/openstreetmap/iD/labels/good%20first%20issue)" label. I think it would be better if all issues that receive this label also get a short comment on how you think this should be solved. (I am aware that some implementation issues will only come up once the work starts, but a general idea on how you would start working on it would still be helpful IMO.)

**Comment by AnitaMari** [comment]
Thank you! I think I didn't choose a "good first issue" to start my journey
in the open-source contribution world. I think I'll look for another one to
try to understand how everything works before trying to solve something
more complex for me to understand.

**Comment by matkoniecz** [comment]
> The deprecation was done already https://github.com/openstreetmap/iD/issues/7338. And it is in line with what the wiki says https://wiki.openstreetmap.org/wiki/Key:created_by So I wonder what this issue is actually about…

See https://wiki.openstreetmap.org/wiki/Discardable_tags 

To be more specific `created_by` tags will be silently and automatically removed by iD if their objects were edited. See https://www.openstreetmap.org/node/269485010/history

Note that I have only moved the node, never edited tags and tag was dropped by iD.

JOSM id doing the same for discardable tags, but it is also not showing them on tag list

**Comment by Adarsh-Dhar** [comment]
Hey its my first time contributing to open source. I want to contribute to this issue can you tell me what do you want instead of created_by

**Comment by 1ec5** [comment]
Even if the tags are removed upon editing the feature, are we sure it’s a good idea to not even show them in the _raw_ tag editor, as if they never existed? If the user decides not to edit the feature, this might cause them to be confused about whether they removed the tags. Graying out the fields might be less confusing – along with adding a tooltip explaining why.

**Comment by matkoniecz** [comment]
> If the user decides not to edit the feature

in such case tags will not be autoremoved

**Comment by matkoniecz** [comment]
In fact, if tags are shown during editing and then deleted automatically with no user interaction then it is more confusing for user

**Comment by Adarsh-Dhar** [comment]
So do you want to remove the tag?

**Comment by matkoniecz** [comment]
this tags are already being automatically removed on edit

what I proposed here is to stop displaying them in tag view

**Comment by 1ec5** [comment]
>> If the user decides not to edit the feature
> 
> in such case tags will not be autoremoved

I understand. My point is that this would be potentially just as confusing as:

> In fact, if tags are shown during editing and then deleted automatically with no user interaction then it is more confusing for user

It’s literally the reverse situation. So I think a more user-friendly and possibly more straightforward change would be to style the discardable tags differently. There’s no way to style individual tags in the text view, but that’s a raw, advanced representation of a raw, advanced representation. If that’s a showstopper, I wonder what practical problem it would solve.

Here’s more or less where an additional HTML class would be set if the key is discardable:

https://github.com/openstreetmap/iD/blob/407360943ce5db9d98e857f6148c7876d789ff37/modules/ui/sections/raw_tag_editor.js#L159

The styling could be similar to the styling for tags in a read-only view, maybe with the addition of `text-decoration: line-through`:

https://github.com/openstreetmap/iD/blob/407360943ce5db9d98e857f6148c7876d789ff37/css/80_app.css#L2653-L2661

**Comment by matkoniecz** [comment]
> If that’s a showstopper, I wonder what practical problem it would solve.

some people wanted to run bot edit removing `created_by` to stop confusing display in iD (including raw tag display)

(just running bot edit and removing `created_by` is also an option, I guess)

**Comment by Adarsh-Dhar** [comment]
I know it might be a silly question. But can anyone help me how do I recreate this problem through my local device.

I have already cloned this to my editor and now don't know what to do

**Comment by matkoniecz** [comment]
>   tyrasd added `usability` `good first issue`

so it is accepted as a good idea and just waits for someone to implement it, right?

**Comment by KrishavRajSingh** [comment]
is this open?

**Comment by matkoniecz** [comment]
>is this open? 

yes

![screen](https://github.com/openstreetmap/iD/assets/899988/a9ddc224-0cda-4e3c-8eb8-81568fd4eed1)

**Comment by gy-mate** [comment]
> > tyrasd added `usability` `good first issue`
> 
> so it is accepted as a good idea and just waits for someone to implement it, right?

@tyrasd?

**Comment by 1ec5** [comment]
> just running bot edit and removing created_by is also an option, I guess

Yes, it is an option and a better alternative to your proposal, in my opinion. If these tags are so harmful that users shouldn’t see them, then a bulk edit would prevent these tags from appearing on osm.org and in other editors. On the other hand, if that would be too drastic, then iD can simply style the tags differently, as I described above. Keep in mind that users are more likely to see these tags on osm.org than in iD’s raw tag editor, which is hidden by default.

**Comment by tyrasd** [comment]
> So I think a more user-friendly and possibly more straightforward change would be to style the discardable tags differently. 

I think this would be the best approach. I'd also add a tooltip to explain why the respective tags are "grayed out".
For the "text view" tag editor, there could be a tooltip that shows the discardable tags, if there are any.

**Comment by matkoniecz** [comment]
with 1000+ open issues and this being low priority it makes no sense to keep it open
